### PR TITLE
Ft  preserve data

### PR DESCRIPTION
**Pull request by Bateyjosue**
# FEATURE DESCRIPTION

## TASK

In this project, we preserve the data from our app and load preserved data on startup.

## Cheers

# Happy Review

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -134,4 +138,11 @@ def message
     puts '7 - Exit'
```
Comment (HAMMAS-SALEEM) [inline app.rb:162]: image

- [ ] After creating books or person or rentals when we quit the application in the terminal then it shows this error and doesn't make any change in the data json files :fearful: Kindly fix this issue :blush:

File: app.rb [hunk 2]
```diff
@@ -3,13 +3,15 @@
...
   attr_accessor :persons, :books, :rentals
 
   def initialize()
-    @persons = []
-    @books = []
-    @rentals = []
+    @persons = [JSON.parse(File.read('persons.json'))]
+    @books = [JSON.parse(File.read('books.json'))]
+    @rentals = [JSON.parse(File.read('rentals.json'))]
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: image

- [ ] When we start the application without having data json files then your application shows this error shown above :arrow_up: so your application should see that if book.json file or rental.json or person.json file doesn't exist then your application should create these files before initializing the application :blush:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @AlierPM, @Bateyjosue :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following GitFlow
✔️ Following some requirements

Good job on implementing some of the reqirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
Hi @Bateyjosue,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**